### PR TITLE
Fix WebSocket Hanging issue with Workers

### PR DESCRIPTION
**Pull request by irunika**
This PR fixes the issue of hanging the handshake when using workers inside WebSocket resources.

### CODE REVIEW

File: modules/ballerina-http/src/main/java/org/ballerinalang/net/ws/BallerinaWsServerConnectorListener.java [hunk 1]
```diff
@@ -56,6 +58,8 @@ public void onMessage(WebSocketInitMessage webSocketInitMessage) {
         WebSocketService wsService = WebSocketDispatcher.findService(webSocketInitMessage);
         Resource onHandshakeResource = wsService.getResourceByName(Constants.RESOURCE_NAME_ON_HANDSHAKE);
         if (onHandshakeResource != null) {
+            Semaphore semaphore = new Semaphore(0);
+            AtomicBoolean isResourceExeSuccessful = new AtomicBoolean(false);
```
Comment (hasithaa) [inline modules/ballerina-http/src/main/java/org/ballerinalang/net/ws/BallerinaWsServerConnectorListener.java]: why do we need an atomic boolean?

Comment (irunika) [reply modules/ballerina-http/src/main/java/org/ballerinalang/net/ws/BallerinaWsServerConnectorListener.java]: We simply need a single object to change in inner class. Assigning new value inside inner class is not allowed. 
It gives the error "local variables referenced from an inner class must be final or effectively final"
So should I use another way to deal with it?

Comment (hasithaa) [reply modules/ballerina-http/src/main/java/org/ballerinalang/net/ws/BallerinaWsServerConnectorListener.java]: Cant, we use Boolean. ?

Comment (hasithaa) [reply modules/ballerina-http/src/main/java/org/ballerinalang/net/ws/BallerinaWsServerConnectorListener.java]: Ok. got it.

**Pull request merged**